core/auth/payment_metadata.go

```diff
@@ -0,0 +1,168 @@
+package auth
+
+import (
+	"crypto/ecdsa"
+	"fmt"
+	"math/big"
+
+	"github.com/Layr-Labs/eigenda/core"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/math"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/signer/core/apitypes"
+)
+
+/* SUBJECT TO MODIFICATIONS */
+
+// EIP712Signer handles EIP-712 domain specific signing operations over typed and structured data
+type EIP712Signer struct {
+	domain apitypes.TypedDataDomain
+	types  apitypes.Types
+}
+
+// NewEIP712Signer creates a new EIP712Signer instance
+func NewEIP712Signer(chainID *big.Int, verifyingContract common.Address) *EIP712Signer {
```

Comment:
Does the `verifyingContract` ever have a function?

Reply:
Nope, verifyingContract would only affect the output signature and the recovery